### PR TITLE
[styled-components] Replace Omit with loose version

### PR DESCRIPTION
**Pull request by goodmind**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Current definitions error on 
```ts
import styled from 'styled-components'
import {Button} from '@material-ui/core'

const StyledButton = styled(Button)`
`

const test = <StyledButton color="inherit">Click</StyledButton>
```

### CODE REVIEW

File: types/styled-components/index.d.ts [hunk 1]
```diff
@@ -330,7 +330,9 @@ export type ThemedCssFunction<T extends object> = BaseThemedCssFunction<
 >;
 
 // Helper type operators
-type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+type Omit<T, K extends keyof any> = T extends any
```
Comment (Igorbek) [inline types/styled-components/index.d.ts]: I don't really understand why you need to change to this.
The construction doesn't make much sense to me.
`T` always extends `any`, and why the constraints get relaxed?

**Review by Igorbek — CHANGES_REQUESTED** [review]
Thank you for the contribution,
I would like to see an explanation why do you think the solution is better than what we had.

File: types/styled-components/index.d.ts (hunk 1 above)
Comment (goodmind) [reply types/styled-components/index.d.ts]: I just got this Omit from @material-ui/core, because current Omit doesn't work with this library

Comment (goodmind) [reply types/styled-components/index.d.ts]: Here or in styled-components repo? Should I close this pull request?

Comment (Igorbek) [reply types/styled-components/index.d.ts]: Here, in DT, please.

**Pull request closed without merging**